pydm/widgets/image.py

```diff
@@ -18,6 +22,53 @@ class ReadingOrder(object):
     Clike = 1
 
 
+class ImageUpdateThread(QThread):
+    updateSignal = pyqtSignal(list)
+
+    def __init__(self, image_view):
+        QThread.__init__(self)
+        self.image_view = image_view
+
+    def run(self):
+        img = self.image_view.image_waveform
+        needs_redraw = self.image_view.needs_redraw
+        image_dimensions = len(img.shape)
+        width = self.image_view.imageWidth
+        reading_order = self.image_view.readingOrder
+
+        self.image_view.lock.acquire()
```

Comment:
This code could probably be made a little less brittle by using a `with <lock>` statement here, so that you don't have to make sure that you release the lock in every potential exit route.